baybe/constraints/discrete.py

```diff
@@ -287,11 +311,11 @@ def get_invalid(self, data: pd.DataFrame) -> pd.Index:  # noqa: D102
     DiscreteLinkedParametersConstraint,
     DiscreteSumConstraint,
     DiscreteProductConstraint,
+    DiscreteCardinalityConstraint,
```

Comment:
Thanks for raising it. I'm not quite confident with its position. Would appreciate if @Scienfitz @AdrianSosic can give some input.

Reply:
i did not see it as deliberate decision, but I also did not find any arg why it belongs somewhere else, maybe theres some?

Reply:
OK, had a quick call with @Waschenbacher. I think we need some guidance here from you, @Scienfitz. As far as I can remember, the order has generally two purposes
1. (Not a strict requirement, but useful): Apply the constraints in a "reasonable" order to make the filter as fast as possible, i.e. have "cheap" constraints that potentially reduce the space by a large part first
2. Certain constraints have to be applied in a certain order

Unfortunately, I cannot really remember the details of 2, and I think we should also put an explanation next to the declaration of the order, to remind future us.

So can you quickly comment on both 1 and 2, and ideally also write the text for the comment?

Reply:
I think 2) was not applicable to anything in the list so it was mostly 1).

If we order according to expected comp effort the new constraint should prob come after sum/product which is already the case.

Perhaps `CustomConstraint` should then also be moved from first to third-last position.

Comment 
```
# Constraints are approximately ordered according to increasing computational effort to minimize total time in their sequential application
```

Reply:
Oh, OK I wasn't actually aware of that. Could swear that there was some (partial) order required, but if you say it's not, then it seems I'm wrong here. Just wondering why we had that list in the first place then, because I think we never talked about complexity before, and we could have simply iterated over the incoming constraints then ... 🤔 

But I'm happy with your suggest. Can you adjust it accordingly, @Waschenbacher?